src/gtc/passes/oir_optimizations/temporaries.py

```diff
@@ -78,12 +78,13 @@ def visit_VerticalLoop(
 
     def visit_Stencil(self, node: oir.Stencil, **kwargs: Any) -> oir.Stencil:
         tmps_to_replace = kwargs["tmps_to_replace"]
+        all_names = collect_symbol_names(node)
         return oir.Stencil(
             name=node.name,
             params=node.params,
             vertical_loops=self.visit(
                 node.vertical_loops,
-                new_symbol_name=symbol_name_creator(set(kwargs["symtable"])),
+                new_symbol_name=symbol_name_creator(all_names),
```

Comment:
This change is necessary when making HorizontalExecution a SymbolTableTrait. Temporaries could be made that conflicted with existing local names.